test/Common/ServerFixture.cs

```diff
@@ -63,6 +63,8 @@ private void StartServer()
                 _logger.LogInformation("Test server shut down");
                 _logToken.Dispose();
             });
+
+            t.Wait();
```

Comment:
Change all of the code to use the new assemblies.

Reply:
What?

Reply:
aren't we actually waiting already on line 50?

Reply:
Stuff still happens after registered `IApplicationLifetime`s are started.

Reply:
Let's talk about this on Monday. This at least needs a timeout but if _lifetime.ApplicationStarted is not good enough then shouldn't we replace it entirely with waiting on `t` ?

Reply:
Only this happens in a ctor and I am not sure if this can be changed easily since it is an XUnit fixture.

Reply:
We should be able to  remove our custom logic and use the new methods. It should be enough to cover these scenarios.

Reply:
How, the new APIs I see are `Run` and `RunAsync` both of which just call `Start` but give no indication of when `Start` finishes.